Source/ACE/Entity/WorldObject.cs

```diff
@@ -168,11 +168,21 @@ public ObjectDescriptionFlag DescriptionFlags
 
         public WeenieHeaderFlag WeenieFlags
         {
-            get { return (WeenieHeaderFlag)AceObject.WeenieHeaderFlags; }
+            get { return SetWeenieHeaderFlag(); }
             protected internal set { AceObject.WeenieHeaderFlags = (uint)value; }
         }
 
-        public WeenieHeaderFlag2 WeenieFlags2 { get; protected set; }
+        public WeenieHeaderFlag2 WeenieFlags2
+        {
+            get
+            {
+                WeenieHeaderFlag2 flags = SetWeenieHeaderFlag2();
+                if (flags != WeenieHeaderFlag2.None)
+                    IncludesSecondHeader = true;
```

Comment:
The logic in these setters is getting slightly complex for my tastes (not necessarily a problem, if its essential), but its something you should think about.  Don't want setting a property to have unintended/unexpected consequences for those unfamiliar with the code.